### ISSUE TITLE
Support for CockroachDB serverless

### ISSUE THREAD

**Issue by ashsmith** [prisma/prisma#9978]
## Problem

As it stands Prisma ignores the `options` parameter that is in CockroachDB's connection string for their serverless/cloud plans this value indicates which cluster to connect to. Without this value you'll recieve a connection error.

Valid values for the `options` paramter for cockroach is:

```
options=--cluster=%3Dmy-cluster-name
or:
options=-c%3Dmy-cluster-name
```

This is documented here: https://www.cockroachlabs.com/docs/stable/connection-parameters.html#supported-options-parameters

A connection string will look similar to this:

```
postgresql://{{USER}}:{{PASS}}@{{HOST}}:{{PORT}}/defaultdb?options=--cluster%3D{{CLUSTER_NAME}}
```

## Suggested solution

Support the `options` parameter in the postgres connection string.

## Additional context

I receive the following error:

```
Database connection error:

undefined: Database error
Error querying the database: db error: FATAL: codeParamsRoutingFailed: missing cluster name in connection string
```

**Comment by janpio** [comment]
All that being said, the cluster name in the connection string can be worked around:

> To construct a suitable connection string, open the cluster in the CockroachCloud, click the `Connect` button on the rop right to open the `Connection info` modal. There go to the `Connection parameters` tab and construct a connection string with these values. 

https://github.com/prisma/e2e-tests/pull/1597/files#diff-3ff81f6c96e193d32a22f6cea59f0e08e0f4609c685d3e0c8f2bd0346e8abd49R7

That way should be able to connect to the database. Let me know if that works @ashsmith

**Comment by janpio** [comment]
@ashsmith @otan Can you please confirm that using the individual Connection parameters allows you to proceed further? Thanks.

**Comment by alvis** [comment]
@janpio No. I can confirm that the issue @otan reported remains. I still got the following error message

```
Error: db error: ERROR: unknown function: pg_get_functiondef()
   0: migration_core::api::SchemaPush
             at migration-engine/core/src/api.rs:184
```

**Comment by janpio** [comment]
@alvis You just confirmed that you could actually connect to the server using the Connection Parameters from the Cockroach UI (vs. the `Error querying the database: db error: FATAL: codeParamsRoutingFailed: missing cluster name in connection string` error message that @ashsmith initially got), and that you are now getting the _expected_ error message when using `db push` (or another Migrate command) which is an overall problem with CockroachDB and Prisma. We will work on that via https://github.com/prisma/prisma/issues/4702

(Maybe post your current error message and the command you are trying to run in that issue so the current state is reflected there. Thanks!)

**Comment by alvis** [comment]
@janpio Thanks for your quick feedback. Yes. You're right that I got the error from `prisma db push`. The error does indicate that prisma can connect to the database, but it hits a problem that cockroachdb lacks `pg_get_functiondef` for getting info about user-defined functions.

**Comment by janpio** [comment]
"Yes", but it is quite an effort: You can use `db push` (or any other Prisma Migrate command) with a Postgres database, then download the SQL to create the database manually (or take it from the generated migration files) and use it on your CockroachDB server - the end result should be the same. The problem is just in Introspection and internal assumption of Migrations that do not work on CockroachDB.

**Comment by otan** [comment]
For posterity, I've put an issue up on our side detailing what needs to change in either prisma and/or Cockroach to get `migrate` working: https://github.com/cockroachdb/cockroach/issues/73137

We'll make it work together one day :)!

**Comment by janpio** [comment]
Follow up issue: https://github.com/prisma/prisma/issues/10494

**Comment by janpio** [comment]
Update: In recent `dev` versions of Prisma CLI the connection string com CockroachDB Cloud Serverless can just be copy pasted into Prisma and works. No additional work is need any more. This will be released with 3.8.0.
